gp-nested-forms/gpnf-map-child-entries-to-acf-repeater-field.php

```diff
@@ -3,36 +3,53 @@
-// Update "123" to the ID of the Parent form.
-add_action( 'gform_after_submission_123', 'gw_child_entries_to_repeater', 10, 2 );
-function gw_child_entries_to_repeater( $entry, $form ) {
 
-	// Update "4" to the ID of the Nested Form field.
-	$nested_form_field_id = 4;
-	// The field map contains "field_name" => "child_entry_field_id" pairs. The field name is the name of the fields in
-	// the ACF Repeater field. The child entry field ID is the field ID from the child form.
-	$field_map = array(
-		'custom_field_one' => 5,
-		'custom_field_two' => 6,
-	);
-	// Update test_repeat to the field name of the ACF repeater field.
-	$acf_repeater_field_name = 'test_repeat';
-	/* STOP! You don't need to edit below this line. */
+class GPNF_Map_Child_Entries_To_ACF_Repeater {
+
+	private $args;
+
+	public function __construct( $args ) {
+		$this->_args = wp_parse_args( $args, array(
+			'form_id'                 => false,
+			'nested_form_field_id'   => false,
+			'field_map'               => array(),
+			'acf_repeater_field_name' => false,
+		) );
+
+		add_action( 'gform_advancedpostcreation_post_after_creation', array( $this, 'gw_child_entries_to_repeater' ), 10, 2 );
+	}
+
+	function gw_child_entries_to_repeater( $post_id, $feed, $entry, $form ) {
 
-	$parent_entry  = new GPNF_Entry( $entry );
-	$child_entries = $parent_entry->get_child_entries( $nested_form_field_id );
-	$repeat_value  = array();
-	$created_posts = gform_get_meta( $entry['id'], 'gravityformsadvancedpostcreation_post_id' );
-	foreach ( $created_posts as $post ) {
-		$post_id = $post['post_id'];
-		foreach ( $child_entries as $child_entry ) {
-			$value = array();
-			foreach ( $field_map as $acf_field_name => $child_entry_field_id ) {
-				$value[ $acf_field_name ] = rgar( $child_entry, $child_entry_field_id );
+		if ( $form['id'] !== $this->_args['form_id'] ) {
+			return;
+		}
+	
+		$parent_entry  = new GPNF_Entry( $entry );
+		$child_entries = $parent_entry->get_child_entries( $this->_args['nested_form_field_id'] );
+		$repeat_value  = array();
+		$created_posts = gform_get_meta( $entry['id'], 'gravityformsadvancedpostcreation_post_id' );
+		foreach ( $created_posts as $post ) {
+			$post_id = $post['post_id'];
+			foreach ( $child_entries as $child_entry ) {
+				$value = array();
+				foreach ( $this->_args['field_map'] as $acf_field_name => $child_entry_field_id ) {
+					$value[ $acf_field_name ] = rgar( $child_entry, $child_entry_field_id );
+				}
+				array_push( $repeat_value, $value );
 			}
-			array_push( $repeat_value, $value );
+			update_field( $this->_args['acf_repeater_field_name'], $repeat_value, $post_id );
 		}
-		update_field( $acf_repeater_field_name, $repeat_value, $post_id );
 	}
+ }
 
-}
+ new GPNF_Map_Child_Entries_To_ACF_Repeater( array(
+	'form_id'                  => 7, // Set this to the parent form ID
+	'nested_form_field_id'     => 18, // Update to the ID of the Nested Form field.
+	'field_map'                => array(
+		'nom_comedien'         => 1,
```

Comment:
```suggestion
		'nom_comedian'         => 1,
```

Not a blocker, but figured I'd point out this typo. It's in a few spots.

Should this also be `num` instead of `nom`? I just want to make sure this example makes sense.